### ISSUE TITLE
Connector not pushing data to Kafka

### ISSUE THREAD

**Issue by rajilion** [jpbirdy/kafka-hdfs-source-connector#1]
name=test-hdfs
connector.class=hdfs.HDFSSourceConnector
hadoop.conf.dir=/etc/hadoop/conf
hadoop.home=
hdfs.url=hdfs://nameservice1
hdfs.authentication.kerberos=true
connect.hdfs.principal=principal
connect.hdfs.keytab=keytab
hdfs.namenode.principal=principal
kerberos.ticket.renew.period.ms=3600000
file=
file.path=/user/directorypath
topic.prefix=test-hdfs-source

i am using the above config to connect hdfs and push data from directory to kafka

from the logs i could see the connection has been established with hadoop as below

[2018-08-01 08:47:56,013] INFO Created connector test-hdfs (org.apache.kafka.connect.cli.ConnectStandalone:99)
[2018-08-01 08:47:56,013] INFO Hadoop configuration directory /etc/hadoop/conf (hdfs.HDFSSourceTask:79)
[2018-08-01 08:47:56,272] INFO Hadoop namenode principal: hdfs/_HOST@DOMAIN.COM (hdfs.HDFSSourceTask:299)
[2018-08-01 08:47:56,273] INFO Hadoop namenode principal: hdfs/_HOST@DOMAIN.COM (hdfs.HDFSSourceTask:299)
[2018-08-01 08:47:56,609] INFO Login successful for user principal using keytab file principal.keytab (org.apache.hadoop.security.UserGroupInformation:966)
[2018-08-01 08:47:56,655] INFO Login successful for user principalusing keytab file principal.keytab (org.apache.hadoop.security.UserGroupInformation:966)
[2018-08-01 08:47:56,656] INFO Login as: principal (hdfs.HDFSSourceTask:304)
[2018-08-01 08:47:56,656] INFO Login as: principal (hdfs.HDFSSourceTask:304)
[2018-08-01 08:47:56,657] INFO Starting the Kerberos ticket renew thread with period 3600000ms. (hdfs.HDFSSourceTask:334)
[2018-08-01 08:47:56,657] INFO Starting the Kerberos ticket renew thread with period 3600000ms. (hdfs.HDFSSourceTask:334)

also i can see offset loading information as below

[2018-08-01 08:47:58,043] INFO loading offset /user/directory/e24f6ce5225ebcd6-31762fb100000004_2144770316_data.0. (hdfs.HDFSSourceTask:258)
[2018-08-01 08:47:58,073] INFO loading offset /user/directory/e24f6ce5225ebcd6-31762fb100000004_1165567777_data.0. (hdfs.HDFSSourceTask:258)

but i when i run console consumer for the topic i could hardly receive any message.

can you please help me solving this? 

Thanks in advance

**Comment by jpbirdy** [comment]
Maybe something wrong with your Kafka, not this connector. I'm sure it can push messages from HDFS to Kafka.

**Comment by rajilion** [comment]
Yes it pushes record thank you😊. Just made few modifications to commit each records to Kafka since committing as whole resulted in Java outofmemory error and to convert avro file to struct

**Comment by Nawter** [comment]
Hi @rajilion and @jpbirdy I had the same problem I try to send files from HDFS to local, or from HDFS to s3 but I can't see this statements in my logs
```
[2018-08-01 08:47:58,043] INFO loading offset /user/directory/e24f6ce5225ebcd6-31762fb100000004_2144770316_data.0. (hdfs.HDFSSourceTask:258)
[2018-08-01 08:47:58,073] INFO loading offset /user/directory/e24f6ce5225ebcd6-31762fb100000004_1165567777_data.0. (hdfs.HDFSSourceTask:258)
```

**Comment by Nawter** [comment]
This the end of my logs
```
[2019-05-28 15:19:42,499] INFO test.txt, 啥也没读到,让我歇一时 (hdfs.HDFSSourceTask:129)
[2019-05-28 15:19:42,499] INFO /user/almalikz, 啥也没读到,让我歇一时 (hdfs.HDFSSourceTask:129)
```

**Comment by jpbirdy** [comment]
@Nawter You can see https://github.com/jpbirdy/kafka-hdfs-source-connector/blob/master/src/main/java/hdfs/HDFSSourceTask.java#L128-L131. If you get this logs, it means connector cannot read any record in this file.

**Comment by Nawter** [comment]
@jpbirdy why is that, how can I debug this code.
Cheers.

**Comment by Nawter** [comment]
java.lang.IllegalArgumentException: java.net.UnknownHostException: nameservice1, I have this problem when I run it and is from this line 
```
public List<SourceRecord> poll() throws InterruptedException {
    try {
      fileSystem = FileSystem.get(URI.create(hdfsUrl), conf);
```
Cheers
